### PR TITLE
Retrieve fallback schedule at different location (url) if needed

### PR DESCRIPTION
**Pull request by Flix6x**
Implement logic to retrieve a fallback schedule at a different location (url).

### CODE REVIEW

File: flexmeasures_client.py [hunk 1]
```diff
@@ -176,6 +176,16 @@ def get_schedule(self, kwargs, **fnc_kwargs):
             headers={"Authorization": self.fm_token},
         )
         self.check_deprecation_and_sunset(url, res)
+        if res.status_code == 303):
+            new_url = res.headers.get("location")
```
Comment (ArdJonker) [inline flexmeasures_client.py:180]: Does the returned location include a (new) schedule_id or is this a generic fallback URL?

**Review by ArdJonker — COMMENTED** [review]
Is this a breaking change in the API that all users need to upgrade to?

File: flexmeasures_client.py (hunk 1 above)
Comment (Flix6x) [reply flexmeasures_client.py:180]: Yes, the location includes the new schedule id.

**Comment by Flix6x** [comment]
> Is this a breaking change in the API that all users need to upgrade to?

Yes, this PR prepares API clients for a breaking change in the API of the FlexMeasures server when hosts decide to upgrade their server to `flexmeasures==0.17`.

**Review by ArdJonker — APPROVED** [review]

File: flexmeasures_client.py (hunk 1 above)
Comment (Flix6x) [reply flexmeasures_client.py:180]: @victorgarcia98 For API clients, does the location return the full url or only the part with `/api/v3_0/sensors/<id>/schedules/<uuid>`, as we test [here](https://github.com/FlexMeasures/flexmeasures/blob/3a9d52430973bf6fcc6b8c604aba00b8baf7d3aa/flexmeasures/api/v3_0/tests/test_sensor_schedules.py#L443-L447)? The location url is created [here](https://github.com/FlexMeasures/flexmeasures/blob/3a9d52430973bf6fcc6b8c604aba00b8baf7d3aa/flexmeasures/api/v3_0/sensors.py#L475-L481) using `flask.url_for`.

Comment (GustaafL) [reply flexmeasures_client.py:180]: From testing it only returns a partial url

Comment (Flix6x) [reply flexmeasures_client.py:180]: https://github.com/FlexMeasures/flexmeasures/pull/888 will make sure the server returns the full url instead, thanks @GustaafL!

**Pull request merged**